rust/oak/src/storage/mod.rs

```diff
@@ -60,16 +61,24 @@ impl Storage {
         }
     }
 
-    fn execute_operation(
-        &mut self,
-        operation_request: &StorageChannelRequest,
-    ) -> StorageChannelResponse {
+    fn execute_operation<Req, Res>(&mut self, operation_request: &Req) -> grpc::Result<Res>
```

Comment:
Req/Res seem borderline for immediate understandability.

RequestType, ResponseType or RequestT, ResponseT?

"Do not worry about saving horizontal space as it is far more important to make your code immediately understandable by a new reader. Minimize the use of abbreviations that would likely be unknown to someone outside your project"

https://google.github.io/styleguide/cppguide.html#General_Naming_Rules

Reply:
Well, I generally agree, but these are generic types, for which usually single-letter identifiers are used (T, S). I think making them look like concrete types would actually be more confusing :)